connector/connect/server/src/main/scala/org/apache/spark/sql/connect/service/ExecuteEventsManager.scala

```diff
@@ -193,12 +202,21 @@ case class ExecuteEventsManager(executeHolder: ExecuteHolder, clock: Clock) {
   /**
    * Post @link org.apache.spark.sql.connect.service.SparkListenerConnectOperationFinished.
    */
-  def postFinished(): Unit = {
+  def postFinished(totalNumRowsOpt: Option[Long] = None): Unit = {
```

Comment:
Consider renaming to `producedRowsCountOpt`.
Please add documentation to indicate that this is the number of rows that are returned to the user. None is expected when the command does not return any rows.